activesupport/test/descendants_tracker_test_cases.rb

```diff
@@ -28,12 +28,14 @@ def test_descendants
   end
 
   def test_descendants_with_garbage_collected_classes
-    1.times do
-      child_klass = Class.new(Parent)
-      assert_equal_sets [Child1, Grandchild1, Grandchild2, Child2, child_klass], Parent.descendants
-    end
-    GC.start
-    assert_equal_sets [Child1, Grandchild1, Grandchild2, Child2], Parent.descendants
+    Thread.new do
+      1.times do
+        child_klass = Class.new(Parent)
+        assert_equal_sets [Child1, Grandchild1, Grandchild2, Child2, child_klass], Parent.descendants
+      end
+      GC.start
```

Comment:
```suggestion
    # The Ruby GC (and most other GCs for what matters) are not fully precise.
    # When GC run, the whole stack is scanner to mark any object reference in registers.
    # But some of these references might simply be leftovers from previous method calls
    # waiting to be overriden, and there's no definite way to clear them.
    # By executing this code in a distinct thread, we ensure that such eventual references
    # are on a stack that will be entirely garbage collected, effectively working around the problem.
    Thread.new do
        child_klass = Class.new(Parent)
        assert_equal_sets [Child1, Grandchild1, Grandchild2, Child2, child_klass], Parent.descendants
    end

    # Calling `GC.start` 4 times should trigger a full GC run
    4.times do
      GC.start
    end
```

Reply:
Not sure if the explanation is worth it or not. Either way if we execute in a distinct thread, we no longer need the `1.times` which as far as I can tell was used to create a scope.

And `4.times { GC.start }` should trigger a full GC run, reducing flakiness further.

Reply:
Is calling `GC.start` 4 times an official / semi-official mechanism to trigger a full GC run?  Or is it more like "should hopefully"?

Reply:
Not sure which, but yes it's a well known and used pattern, see nakayoshi_fork for instance:

  - https://github.com/ko1/nakayoshi_fork/commit/70962fd0fd21bc7dab190b7bb2c8bc31cb063c96
  - https://github.com/ko1/nakayoshi_fork#mechanism